src/gpg/contact/delete.py

```diff
@@ -21,10 +21,11 @@ def delete(contacts, force):
                 if ans.lower() == 'y' or force:
                     c.op_delete(key, False)
 
-    except BaseException:
+    except gpg.errors.GpgError as e:
         if os.environ['DEBUG'] == 'yes':
             raise
-        exit(2)
+        print(e, file=sys.stderr, flush=True)
+        exit(1)
```

Comment:
These 4 lines seem to be the same everywhere. Why don't you create a utility (auxiliary) function to encapsulate them, for example: `handle_exception(e)`

Reply:
Also shouldn't the print statement be in the else statement?

Reply:
Umm I guess raise causes an exit as well.